src/lib.rs

```diff
@@ -138,18 +142,68 @@ pub struct RegisteredClaims {
     pub aud: Option<SingleOrMultipleStrings>,
     /// Expiration time in seconds since Unix Epoch
     #[serde(skip_serializing_if = "Option::is_none")]
-    pub exp: Option<u64>,
+    pub exp: Option<i64>,
     /// Not before time in seconds since Unix Epoch
     #[serde(skip_serializing_if = "Option::is_none")]
-    pub nbf: Option<u64>,
+    pub nbf: Option<i64>,
     /// Issued at Time in seconds since Unix Epoch
     #[serde(skip_serializing_if = "Option::is_none")]
-    pub iat: Option<u64>,
+    pub iat: Option<i64>,
     /// JWT ID
     #[serde(skip_serializing_if = "Option::is_none")]
     pub jti: Option<String>,
 }
 
+bitflags! {
+    flags TemporalValidationOptions : u8 {
+        const ISSUED_AT_REQUIRED = 0x1,
+        const NOT_BEFORE_REQUIRED = 0x2,
+        const EXPIRY_REQUIRED = 0x4,
+        const ALL_TIMES_REQUIRED = ISSUED_AT_REQUIRED.bits
+                                 | NOT_BEFORE_REQUIRED.bits
+                                 | EXPIRY_REQUIRED.bits
+    }
+}
+
+impl RegisteredClaims {
+    fn validate_times(&self,
+                      now: Option<DateTime<UTC>>,
+                      options: TemporalValidationOptions)
+                      -> Result<bool, ValidationError> {
```

Comment:
The return type should be `Result<(), Error>`.  All failures return `Err` and you never return an `Ok(false) `